src/test/ui/consts/dyn-type-name.rs

```diff
@@ -0,0 +1,18 @@
+// run-pass
```

Comment:
This isn't really testing anything about `const` is it?

This doesn't even need a ui test, so `library/core/tests/any.rs` might be a better location.